File: sockets-raiz/client_tcp.py
import socket
import time

HOST = '127.0.0.1'  
PORT = 9000      

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.connect((HOST, PORT))
    print(f'Conectado ao servidor TCP Python em {HOST}:{PORT}.\nPressione Ctrl+C para sair.')

    message = "Olá, servidor!"
    s.sendall(message.encode('utf-8'))
    
    try:
        while True:
            msg = input("Digite mensagem (ou 'sair'): ")
            if msg == 'sair':
                break
            s.sendall(msg.encode('utf-8'))

    except KeyboardInterrupt:
        print('Desconectando...')

File: sockets-raiz/client_tcp.js
const net = require('net'); // Importa o módulo net do Node.js para criar um cliente TCP

const HOST = '127.0.0.1';
const PORT = 9000;

const client = new net.Socket();

client.connect(PORT, HOST, () => {
  console.log(`Conectado ao servidor TCP Node.js\nem ${HOST}:${PORT}.\nPressione Ctrl+C para sair.`);
});

process.on('SIGINT', () => { // Intercepta o sinal Ctrl+C
  console.log('Desconectando...');
  client.end();
  process.exit();
});

client.on('close', () => { // Evento disparado quando a conexão é fechada
  console.log('Conexão encerrada.');
});
